File: frontend/src/model/bullshit/LobbyView.ts
export interface LobbyPlayer {
  seat: number;
  name: string | null;
  joined: boolean;
}

export interface LobbyView {
  started: false;
  gameId: string;
  players: LobbyPlayer[];
  hostSeat: number;
  mySeat: number;
  minPlayers: number;
  maxPlayers: number;
  canStart: boolean;
  // Opaque host-selected game options (e.g. claimMode). Carried now; rendering the chosen
  // mode in the lobby is a deferred follow-up that owns BullshitGameScreen.vue.
  options?: Record<string, string>;
}

File: backend/src/main/java/org/kevinkib/cardgames/sessionmanagement/core/application/LobbyView.java
package org.kevinkib.cardgames.sessionmanagement.core.application;

import org.kevinkib.cardgames.game.PlayerId;
import org.kevinkib.cardgames.sessionmanagement.core.domain.SessionGame;

import java.util.List;

/** Generic per-viewer projection of a not-yet-started session (a lobby). Published; no secrets. */
public record LobbyView(
        boolean started,
        String gameId,
        List<LobbyPlayer> players,
        int hostSeat,
        int mySeat,
        int minPlayers,
        int maxPlayers,
        boolean canStart) {

    public record LobbyPlayer(int seat, String name, boolean joined) {
    }

    static LobbyView forViewer(SessionGame lobby, int minPlayers, int maxPlayers, PlayerId viewer) {
        List<LobbyPlayer> players = lobby.seats().stream()
                .map(seat -> new LobbyPlayer(seat.id().id(), seat.name(), seat.isClaimed()))
                .toList();

        boolean canStart = lobby.isHost(viewer) && lobby.claimedCount() >= minPlayers;

        return new LobbyView(
                false,
                lobby.id().uuid().toString(),
                players,
                SessionGame.HOST_SEAT.id(),
                viewer.id(),
                minPlayers,
                maxPlayers,
                canStart);
    }
}
